org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/console/JavaStackTraceHyperlink.java

```diff
@@ -203,6 +234,232 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
+		int innerClasslevel = innerClassLevels(this.generatedLink);
+		int levelTravelled = 0;
+		while (innerClass.length > 0) {
+			for (IType innerType : innerClass) {
+				if (innerClass.length > 0) {
+					innerClass = innerType.getTypes();
+				}
+				levelTravelled++;
+				if (extractFromResults(innerType, methodSignature, methodNameExtracted) && levelTravelled == innerClasslevel) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Checks the levels of inner class
+	 *
+	 * @param genLink
+	 *            Generated link
+	 * @return returns <code>count of levels</code>
+	 */
+
+	private int innerClassLevels(String genLink) {
+		int level = 0;
+		for (Character c : genLink.toCharArray()) {
+			if (c == '$') {
+				level++;
+			}
+		}
+		return level;
+	}
+	/**
+	 * Checks if there's any matching methods for the given IType
+	 *
+	 * @param type
+	 *            Type of the matches
+	 * @param methodSignature
+	 *            entire method declaration
+	 * @param methodNameExtracted
+	 *            method name
+	 * @throws JavaModelException
+	 * @return returns <code>true</code> if a found an exact method inside the class, or <code>false</code> if there's no matching methods
+	 */
+	private boolean extractFromResults(IType type, String methodSignature, String methodNameExtracted) throws JavaModelException {
+		boolean paramDetailsNotIncluded = methodSignature.substring(methodSignature.indexOf('(')).contains(".java:"); //$NON-NLS-1$
+		try {
+			IMethod[] methods = type.getMethods();
+			for (IMethod method : methods) {
+				if (paramDetailsNotIncluded && method.getElementName().equals(methodNameExtracted)) {
+					return true;
+				}
+				String methodDetails = method.toString();
+				int indexOfClosing = methodDetails.indexOf(')');
+				int indexOfStart = methodDetails.indexOf(method.getElementName());
+				String methodName = methodDetails.substring(indexOfStart, indexOfClosing + 1).replaceAll(" ", ""); //$NON-NLS-1$//$NON-NLS-2$
+				int paramCount = methodSignature.substring(methodSignature.indexOf('(')
+						+ 1, methodSignature.lastIndexOf(')')).split(",").length; //$NON-NLS-1$
+				if (methodName.equals(methodSignature)) {
+					return true;
+				} else if (methodNameExtracted.equals(method.getElementName())
+						&& paramCount == method.getNumberOfParameters()) {
+					// Further mining from fully qualified parameter names in method signature
+					String methodSignatureGen = methodSignatureGenerator(method.getElementName(), methodName);
+					if (methodSignatureGen.equals(methodSignature)) {
+						return true;
+					}
+					// If paramters includes innerclass
+					if (methodSignature.indexOf('$') != -1) {
+						String methodSignatureInnerClass = innerClassMethodSignatureGen(methodNameExtracted, methodSignature);
+						if (methodSignatureInnerClass.equals(methodSignatureGen)) {
+							return true;
+						}
+						String paramsExtracted = methodSignature.substring(methodSignature.indexOf('('));
```

Comment:
Honestly speaking, I feel uncomfortable to see this pattern multiple times, where unchecked output of `indexOf()` is passed straight forward to the `substring()` method, which will happily fail with `IndexOutOfBoundsException` if `indexOf()` returns `-1`.

I see there is a guard around that catches `Exception`, so not a blocker now, but I'm not a big fan of such coding style

Reply:
I agree, but I did added a top-level check to handle IndexOutOfBoundsException, and based on my tests, I haven’t encountered the IndexOutOfBoundsException so far, I see that relying on this assumption may not be ideal in all cases. Now I have included additional check in the critical section. Thanks